### PR TITLE
Additional Notification API and behavior changes

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes
Updated the name of the file hosting the MSFNotification to MSFNotification.swift to match.
Updated the name of the UIKit method to show the notification to show(in...) to match the previous name.
Updated the toast style notifications to have the X button show by default with the dismiss action to match previous behavior of toast style notifications.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Notification_DefaultDismiss_Before | Notification_DefaultDismiss_After |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [x] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [x] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [x] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1103)

### CODE REVIEW

**Comment by harrieshin** [comment]
accessibility on dismiss action works fine?

**Review by harrieshin — APPROVED** [review]

**Comment by huwilkes** [comment]
> accessibility on dismiss action works fine?

Yep! VoiceOver reads out "Dismiss, button".

**Comment by harrieshin** [comment]
> > accessibility on dismiss action works fine?
> 
> Yep! VoiceOver reads out "Dismiss, button".

great update the checkmark on your pr description

File: ios/FluentUI/Notification/MSFNotification.swift [hunk 1]
```diff
@@ -26,6 +26,14 @@ import UIKit
         notification = FluentNotification(style: style,
                                           shouldSelfPresent: false)
         super.init(AnyView(notification))
+        notification.state.actionButtonAction = style.shouldAlwaysShowActionButton
```
Comment (laminesm) [inline ios/FluentUI/Notification/MSFNotification.swift:29]: nit: Would it be possible to store the logic after ? and make this a one-line statement for readability?

**Review by laminesm — APPROVED** [review]

**Pull request merged**